packages/@react-aria/color/src/useColorArea.ts

```diff
@@ -295,32 +311,59 @@ export function useColorArea(props: ColorAreaAriaProps, state: ColorAreaState):
           onThumbDown(e.changedTouches[0].identifier);
         }
       })
-  }, keyboardProps, movePropsThumb);
+  }, focusWithinProps, keyboardProps, movePropsThumb);
+
+  let {focusProps: xInputFocusProps} = useFocus({
+    onFocus: () => {
+      focusedInputRef.current = inputXRef.current;
+    }
+  });
+
+  let {focusProps: yInputFocusProps} = useFocus({
+    onFocus: () => {
+      focusedInputRef.current = inputYRef.current;
+    }
+  });
 
   let isMobile = isIOS() || isAndroid();
 
+  function getAriaLabelForChannel(channel: ColorChannel): string {
+    return isMobile ? state.value.getChannelName(channel, locale) : formatMessage('colorPicker');
+  }
+
+  function getAriaValueTextForChannel(channel:ColorChannel) {
+    return (
+      isMobile || valueChangedViaKeyboard.current ?
+      formatMessage('colorNameAndValue', {name: state.value.getChannelName(channel, locale), value: state.value.formatChannelValue(channel, locale)})
+      :
+      [
+        formatMessage('colorNameAndValue', {name: state.value.getChannelName(xChannel, locale), value: state.value.formatChannelValue(xChannel, locale)}),
+        formatMessage('colorNameAndValue', {name: state.value.getChannelName(yChannel, locale), value: state.value.formatChannelValue(yChannel, locale)})
+      ].join(', ')
+    );
+  }
```

Comment:
If I'm understanding this correctly, mobile screenreaders will always only announce the current channel's value and name. What if the screen reader user performs a double tap + drag interaction to move the slider handle, should/could we have the screenreader announce both channel values like it does for mouse?

I dunno if this is possible (we would have to distinguish the screen reader double tap + drag interaction). Also, not sure if a screenreader user would usually perform a double tap + drag interaction instead of changing the value of each individual channel using swipe motions but I would guess it could happen

Reply:
See 13133361ad96325b1d3b90770e22c2c08e0c8e90.

Reply:
I've includes both channels in `aria-valuetext` on mobile, and label both inputs as a "Color Picker" and "2d slider" so that users can understand it to interact differently than a regular slider. Double tap and hold allows the user to drag the thumb. On Talkback, value text for both channels gets announced intermittently while dragging and when dragging stops. With VoiceOver for iOS, the value text will not be announced as the value changes, but if the user single taps after finishing the drag,  value text for both channels will announce.